protocol/binary_test.go

```diff
@@ -669,3 +669,151 @@ func TestStructOfContainers(t *testing.T) {
 
 	checkEncodeDecode(t, wire.TStruct, tests)
 }
+
+func TestBinaryEnvelopeErrors(t *testing.T) {
+	tests := []struct {
+		encoded []byte
+		errMsg  string
+	}{
+		{
+			encoded: []byte{
+				0x80, 0x02, 0x00, 0x01, // version|type:4 = 2 | call
```

Comment:
This doesn't seem right. All the examples I'm seeing in thriftrw python and node have `0x80, 0x02` as the first 2 bytes.

https://github.com/thriftrw/thriftrw-python/blob/master/tests/protocol/test_binary.py#L405
https://github.com/thriftrw/thriftrw-node/blob/master/test/message.js#L89-90